### PR TITLE
Trips

### PR DESCRIPTION
**Pull request by CHUKWUKA-EMI**
Worked on the trips page and added token to context.

### CODE REVIEW

File: src/Components/Auth/auth.js
Comment (chialuka) [inline src/Components/Auth/auth.js]: why do you have commented code?

File: src/Components/Modal/Modal-Items/ModalItems.js
Comment (chialuka) [inline src/Components/Modal/Modal-Items/ModalItems.js]: any reason for an inline style when you have a CSS file?

File: src/Components/Modal/Modal-Items/ModalItems.js
Comment (chialuka) [inline src/Components/Modal/Modal-Items/ModalItems.js]: why inline style here?

File: src/Components/Modal/modal.js
Comment (chialuka) [inline src/Components/Modal/modal.js]: remove commented code

File: src/Components/Modal/modal.js
Comment (chialuka) [inline src/Components/Modal/modal.js:15]: if there's nothing in the dependency array then there is no need for it here

File: src/Components/Modal/modal.js
Comment (chialuka) [inline src/Components/Modal/modal.js]: since you're already mapping, why create the same component 5 times? You can pass in the values dynamically iinstead

File: src/Components/Modal/modal.js
Comment (chialuka) [inline src/Components/Modal/modal.js]: how do you intend to handle this error from the perspective of the user? How will they know auth has failed?

File: src/Components/NavBar/NavBar.js
Comment (chialuka) [inline src/Components/NavBar/NavBar.js:7]: Your formatting is not standard. Please use 2 tabs and not four across all your files

File: src/Components/Trips/Trips.js
Comment (chialuka) [inline src/Components/Trips/Trips.js]: why is `useState` imported in this component?

File: src/Components/Modal/modal.js
Comment (chialuka) [inline src/Components/Modal/modal.js]: can you create a helper function that will handle this fetching? It's not necessary in this component

File: src/Components/Modal/modal.js
Comment (chialuka) [inline src/Components/Modal/modal.js]: please put this in a separate file and import it here. it doesn't have to be here

File: src/Components/Auth/auth.js
Comment (chialuka) [inline src/Components/Auth/auth.js]: why exactly are you trimming these strings to check length? does empty space count for length?

File: src/Components/Auth/auth.js
Comment (chialuka) [inline src/Components/Auth/auth.js]: is `requestBody` reassigned? If not why are you using let?

File: src/App.js
Comment (chialuka) [inline src/App.js]: please include handling for 404 pages

File: src/Components/Auth/auth.js
Comment (chialuka) [inline src/Components/Auth/auth.js]: how do you ensure that the token isn't lost when the user refreshes the page?

File: src/Components/NavBar/NavBar.css
Comment (chialuka) [inline src/Components/NavBar/NavBar.css:61]: And other screen sizes?

Comment (chialuka) [reply src/Components/NavBar/NavBar.css:61]: you should adopt a mobile-first approach to styling

File: src/Components/NavBar/NavBar.js
Comment (chialuka) [inline src/Components/NavBar/NavBar.js:28]: so logout is always visible, whether there is a token or not?

File: src/App.js
Comment (rowlandekemezie) [inline src/App.js]: Create a custom protectedRoutes for all routes that requires user to be authenticated 

Something like this

```js

const PrivateRoute = ({ component: Component, ...rest }) => (
  <Route {...rest} render={(props) => (
     token
      ? <Component {...props} />
      : <Redirect to={{
          pathname: '/auth',
          state: { from: props.location }
        }} />
  )} />
)

// Then replace the routes with 
<PrivateRoute path="/bookings" component ... />
```

File: src/Components/Modal/modal.css
Comment (rowlandekemezie) [inline src/Components/Modal/modal.css:28]: why do you use percentage? I think you should be using **rem** for now instead

File: src/Components/Modal/modal.js
Comment (rowlandekemezie) [reply src/Components/Modal/modal.js]: you can add that to src/utils/index.js

```js
function makeRequest({url = REACT_APP_API_URL, method='post', data}) {
  return fetch(url, { 
      method,
      body: JSON.stringify(data),
      headers:   {
           "Content-Type": "application/json"
       }
    })
    .then(handleResponse)
    .catch(handleError)
}

// create the function for **handleResponse** and **handleError**
// Add REACT_APP_API_URL=https://ride-booking-app.herokuapp.com/graphql to your environment variable( **.env**)

Then any where you need to make call like here
makeRequest({ data: requestedBody }).....
```

File: src/Components/Modal/modal.js
Comment (rowlandekemezie) [inline src/Components/Modal/modal.js]:
```suggestion
	return routes.map((route) => {
```
You can access each route like this ^^^ and then you wouldn't need hard coded index refering
All you'd need is {route.id}

File: src/Components/Modal/modal.js
Comment (rowlandekemezie) [inline src/Components/Modal/modal.js]:
```suggestion
					id={route.id}
					direction={route.direction}
					duration={route.duration}
					vehicle={route.vehicle}
					cost={route.cost}
				/>
				<ModalItems
					id={route.id}
					direction={route.direction}
					duration={route.duration}
					vehicle={route.vehicle}
					cost={route.cost}
				/>
				<ModalItems
					id={route.id}
					direction={route.direction}
					duration={route.duration}
					vehicle={route.vehicle}
					cost={route.cost}
				/>
				<ModalItems
					id={route.id}
					direction={route.direction}
					duration={routes.duration}
					vehicle={route.vehicle}
					cost={route.cost}
				/>
				<ModalItems
					id={routes.id}
					direction={route.direction}
					duration={route.duration}
					vehicle={route.vehicle}
					cost={route.cost}
				/>
			</div>
```

**Review by rowlandekemezie — CHANGES_REQUESTED** [review]
Try to format your code. set your vscode to 2 spaces. Then right click inside any file and select format code

File: src/App.js
Comment (rowlandekemezie) [reply src/App.js]: You can read more about this concept here(Don't watch video. Just read) => https://tylermcginnis.com/react-router-protected-routes-authentication/

**Comment by CHUKWUKA-EMI** [comment]
Worked on the trips page

**Comment by CHUKWUKA-EMI** [comment]
Worked on the trips page

**Comment by chialuka** [comment]
You're actually supposed to wait for the reviewers to approve before you merge. For example, you pushed your `.env` file with this new commit and merged it. That's wrong.

Please wait for reviewers to approve especially when they have requested changes to the PR

**Pull request merged**
